packages/react-aria-components/docs/Tree.mdx

```diff
@@ -322,6 +833,108 @@ TreeItem also exposes a `--tree-item-level` CSS custom property, which you can u
+
+function Selectable({children}) {
+  let [isSelected, onChange] = React.useState(false);
+  let selectionMode: SelectionMode = isSelected ? 'multiple' : 'none';
+  return (
+    <ToggleButtonContext.Provider value={{isSelected, onChange}}>
+      {/*- begin highlight -*/}
+      <TreeContext.Provider value={{selectionMode}}>
+      {/*- end highlight -*/}
+        {children}
+      </TreeContext.Provider>
+    </ToggleButtonContext.Provider>
+  );
+}
+```
+
+The `Selectable` component can be reused to make the selection mode of any nested `Tree` controlled by a `ToggleButton`.
+
+```tsx example
+import {ToggleButton} from 'react-aria-components';
+
+<Selectable>
+  <ToggleButton>Select</ToggleButton>
+  <PokemonEvolutionTree />
+</Selectable>
+```
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show CSS</summary>
+
+```css
+.react-aria-ToggleButton {
+  margin-bottom: 8px;
+}
+```
+</details>
+
+
+### Custom children
+
+Tree passes props to its child components, such as the selection checkboxes, via their associated contexts. These contexts are exported so you can also consume them in your own custom components. This enables you to reuse existing components from your app or component library together with React Aria Components.
+
+<ContextTable components={['Checkbox', 'Button']} docs={docs} />
+
+This example consumes from `CheckboxContext` in an existing styled checkbox component to make it compatible with React Aria Components. The <TypeLink links={docs.links} type={docs.exports.useContextProps} /> hook merges the local props and ref with the ones provided via context by Tree. See [useCheckbox](useCheckbox.html) for more details about the hooks used in this example.
+
+```tsx
+import type {CheckboxProps, useContextProps} from 'react-aria-components';
+import {CheckboxContext} from 'react-aria-components';
+import {useToggleState} from 'react-stately';
+import {useCheckbox} from 'react-aria';
+
+const MyCustomCheckbox = React.forwardRef((props: CheckboxProps, ref: React.ForwardedRef<HTMLInputElement>) => {
+  // Merge the local props and ref with the ones provided via context.
+  ///- begin highlight -///
+  [props, ref] = useContextProps(props, ref, CheckboxContext);
+  ///- end highlight -///
+
+  let state = useToggleState(props);
+  let {inputProps} = useCheckbox(props, state, ref);
+  return <input {...inputProps} ref={ref} />;
+});
+```
+
+Now you can use `MyCustomCheckbox` within a `Tree`, in place of the builtin React Aria Components `Checkbox`.
+
+```tsx
+<Tree>
+  <TreeItem>
+    <TreeItemContent>
+      {/*- begin highlight -*/}
+      <MyCustomCheckbox slot="selection" />
+      {/*- end highlight -*/}
+      {/* ... */}
+    </TreeItemContent>
+  </TreeItem>
+</Tree>
+```
+
+### Hooks
+
+If you need to customize things even further, such as accessing internal state or customizing DOM structure, you can drop down to the lower level Hook-based API. See [useTree](useTree.html) for more details.
```

Comment:
Looks like useTree docs don't exist. Should also be useTreeGridList also?

Reply:
Yeah, this is when I messaged you about how we don't have those docs yet, so I'll work on creating them next. For now I'll just comment this section out